step/step.go

```diff
@@ -124,34 +134,105 @@ func updateVersionNumbersInProject(helper *projectmanager.ProjectHelper, targetN
 	return nil
 }
 
-func updateVersionNumbersInInfoPlist(helper *projectmanager.ProjectHelper, targetName, configuration string, bundleVersion int64, shortVersion string) error {
+func updateVersionNumbersInInfoPlist(helper *projectmanager.ProjectHelper, schemeName, targetName, configuration string, bundleVersion int64, shortVersion string) error {
 	buildConfig, err := buildConfiguration(helper, targetName, configuration)
 	if err != nil {
 		return err
 	}
 
-	infoPlistPath, err := buildConfig.BuildSettings.String("INFOPLIST_FILE")
+	infoPlistPath, err := buildConfig.BuildSettings.String(infoPlistFileKey)
 	if err != nil {
 		return err
 	}
 
-	absoluteInfoPlistPath := filepath.Join(filepath.Dir(helper.XcProj.Path), infoPlistPath)
+	// By default, the setting for the Info.plist file path is a relative path from the project file. Of course,
+	// developers can override this with something more custom to their setup. They can also use Xcode env vars as part
+	// of their path.
+	//
+	// An example from a SWAT ticket: `$(SRCROOT)/path/to/Info.plist`.
+	//
+	// The problem with this is that it is not a real path until the env var is resolved. And in this case, Xcode
+	// will define this env var, so we only know its value during an xcodebuild execution. So if we see an env var in
+	// the path, then we have to list the build settings with `xcodebuild -showBuildSettings` to get a valid path value.
+	if hasEnvVars(infoPlistPath) {
+		infoPlistPath, err = extractInfoPlistPathWithXcodebuild(helper.XcProj.Path, schemeName, targetName, configuration)
```

Comment:
Can we log a message when this code path runs? Just to explain that an env var is detected, it needs to be resolved by calling Xcode, and it might take a while.

Reply:
Great idea. This will help debugging a lot.